packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -235,6 +235,10 @@ export class CurlyComponentDefinition extends ComponentDefinition {
     builder.attrs.static('class', 'ember-view');
   }
 
+  getComponentClass() {
```

Comment:
1. I don't understand the relationship between caching the layout (I'm not explicitly setting the layout here, just the class) and changing the class
2. Does "static lookup" just mean using a plain `CurlyComponentSyntax`?
3. What is the "dynamic helper"?
4. What's the difference between `DynamicComponentSyntax` and late bound `CurlyComponentSyntax`?